images/virt-artifact/patches/021-support-qcow2-for-filesystem.patch

```diff
@@ -0,0 +1,128 @@
++	cmd := exec.Command("qemu-img", "create", "-f", "qcow2", fullPath, fmt.Sprintf("%db", size))
++	if err = cmd.Run(); err != nil {
++		return fmt.Errorf("failed to create qcow2: %w", err)
++	}
++	return nil
++}
++
+ func getPVCDiskImgPath(volumeName string, diskName string) string {
+ 	return path.Join(pvcBaseDir, volumeName, diskName)
+ }
+@@ -236,7 +246,7 @@ func (hdc *DiskImgCreator) mountHostDiskAndSetOwnership(vmi *v1.VirtualMachineIn
+ 		return err
+ 	}
+ 	if !fileExists {
+-		if err := hdc.handleRequestedSizeAndCreateSparseRaw(vmi, diskDir, diskPath, hostDisk); err != nil {
++		if err := hdc.handleRequestedSizeAndCreateQcow2(vmi, diskDir, diskPath, hostDisk); err != nil {
+ 			return err
+ 		}
+ 	}
+@@ -248,7 +258,7 @@ func (hdc *DiskImgCreator) mountHostDiskAndSetOwnership(vmi *v1.VirtualMachineIn
+ 	return nil
+ }
+ 
+-func (hdc *DiskImgCreator) handleRequestedSizeAndCreateSparseRaw(vmi *v1.VirtualMachineInstance, diskDir string, diskPath string, hostDisk *v1.HostDisk) error {
++func (hdc *DiskImgCreator) handleRequestedSizeAndCreateQcow2(vmi *v1.VirtualMachineInstance, diskDir string, diskPath string, hostDisk *v1.HostDisk) error {
+ 	size, err := hdc.dirBytesAvailableFunc(diskDir, hdc.minimumPVCReserveBytes)
+ 	availableSize := int64(size)
+ 	if err != nil {
+@@ -261,9 +271,9 @@ func (hdc *DiskImgCreator) handleRequestedSizeAndCreateSparseRaw(vmi *v1.Virtual
+ 			return err
+ 		}
+ 	}
+-	err = createSparseRaw(diskPath, requestedSize)
++	err = createQcow2(diskPath, requestedSize)
+ 	if err != nil {
+-		log.Log.Reason(err).Errorf("Couldn't create a sparse raw file for disk path: %s, error: %v", diskPath, err)
++		log.Log.Reason(err).Errorf("Couldn't create a qcow2 file for disk path: %s, error: %v", diskPath, err)
+ 		return err
+ 	}
+ 	return nil
+diff --git a/pkg/virt-launcher/virtwrap/converter/converter.go b/pkg/virt-launcher/virtwrap/converter/converter.go
+index c64b315d8f..71bbe50037 100644
+--- a/pkg/virt-launcher/virtwrap/converter/converter.go
++++ b/pkg/virt-launcher/virtwrap/converter/converter.go
+@@ -590,7 +590,6 @@ func Add_Agent_To_api_Channel() (channel api.Channel) {
+ }
+ 
+ func Convert_v1_Volume_To_api_Disk(source *v1.Volume, disk *api.Disk, c *ConverterContext, diskIndex int) error {
+-
+ 	if source.ContainerDisk != nil {
+ 		return Convert_v1_ContainerDiskSource_To_api_Disk(source.Name, source.ContainerDisk, disk, c, diskIndex)
+ 	}
+@@ -733,7 +732,7 @@ func Convert_v1_Hotplug_DataVolume_To_api_Disk(name string, disk *api.Disk, c *C
+ // Convert_v1_FilesystemVolumeSource_To_api_Disk takes a FS source and builds the domain Disk representation
+ func Convert_v1_FilesystemVolumeSource_To_api_Disk(volumeName string, disk *api.Disk, volumesDiscardIgnore []string) error {
+ 	disk.Type = "file"
+-	disk.Driver.Type = "raw"
++	disk.Driver.Type = "qcow2"
+ 	disk.Driver.ErrorPolicy = "stop"
+ 	disk.Source.File = GetFilesystemVolumePath(volumeName)
+ 	if !contains(volumesDiscardIgnore, volumeName) {
```

Comment:
So, we change raw to qcow2 and there is no way to use "sparse raw" disks on filesystem? It is a significant change in CDI behaviour and should be documented in patch description in README.md.